### PR TITLE
chore: Optimize docker build oss

### PR DESCRIPTION
**Pull request by gastonfournier**
## About the changes
Reduce the build time of OSS docker image from [~30m](https://github.com/Unleash/unleash/actions/workflows/docker_publish.yaml) to [under 15m](https://github.com/Unleash/unleash/actions/runs/5222180536/jobs/9427342758) 

1. Build frontend outside docker multiplatform.
2. Allow `frontend/build` to be copied to the image by removing this from `.dockerignore`
3. Run with `--ignore-scripts` to avoid building the frontend on the `prepare` script, but this requires us to run all the prepare scripts manually (except the frontend build). 
 
**Note:** we need to build frontend in the `prepare` script to be able to have source code dependencies

## Manual Testing
Manually downloaded from https://hub.docker.com/r/unleashorg/unleash-server/tags?page=1 and compared both `unleash` folders from main and the version built with the new process https://github.com/Unleash/unleash/actions/runs/5223078089/jobs/9429430190#step:5:48

Screenshot from 2023-06-10 21-11-33

No major difference was spotted (only expected changes due to development done in main)

**Command used to extract the contents:** 

```
cd /tmp
mkdir main && cd main
docker pull unleashorg/unleash-server:main-edge-18-alpine
docker export $(docker create unleashorg/unleash-server:main-edge-18-alpine) > container.tar && tar xvf container.tar
mkdir ../new-process && cd ../new-process
docker pull unleashorg/unleash-server:sha-ccac902-18-alpine
docker export $(docker create unleashorg/unleash-server:sha-ccac902-18-alpine) > container.tar && tar xvf container.tar
meld ./unleash ../main/unleash
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -8,9 +8,16 @@ COPY . /unleash
 
 RUN yarn config set network-timeout 300000
 
-RUN yarn install --frozen-lockfile && yarn local:package
+RUN yarn install --frozen-lockfile --ignore-scripts
```
[the remark this answers is not in the document]
Comment (gastonfournier) [reply Dockerfile:11]: Ok, I will

**Review by chriswk — CHANGES_REQUESTED** [review]
I think the comments from Sonatype here make sense. We should probably join the three yarn runs here in a `&&` pipeline. To avoid three temporary images.

File: .github/workflows/docker_publish.yaml [hunk 2]
```diff
@@ -38,11 +39,16 @@ jobs:
...
+        run: |
+          yarn --cwd ./frontend install
+          yarn build:frontend
       - name: Build tag and push image to Docker hub
         uses: docker/build-push-action@v3
         with:
+          context: .
           platforms: linux/amd64,linux/arm64
-          push: true
+          push: ${{ github.event_name != 'workflow_dispatch' }}
```
Comment (gastonfournier) [inline .github/workflows/docker_publish.yaml:51]: This enables us to run tests on this workflow without pushing the image

Comment (chriswk) [reply .github/workflows/docker_publish.yaml:51]: Nice!

**Pull request merged**